### ISSUE TITLE
'Explain' button in Query Workbench produces broken/empty modal on malformed query

### ISSUE THREAD

**Issue by stockholmux** [opendistro-for-elasticsearch/sql#863]
Input an invalid query - example:

`SHOW tables LIKE 1;`

Then click the 'Explain' button you get a strange, empty UI modal:

![Screen Shot 2020-11-27 at 7 55 46 AM](https://user-images.githubusercontent.com/1152927/100461420-02767280-3086-11eb-9b15-1caa8d035ac3.png)

I would expect an error message, not an empty box. I assume this is related #586 .

**Comment by chloe-zh** [comment]
Hi @stockholmux  thank you for reporting this issue, will do the enhancement once we have resources.

**Comment by dai-chen** [comment]
An index not found error is expected in this case. I think Query Workbench has problem showing error message either Run or Explain. Related issue: https://github.com/opendistro-for-elasticsearch/sql/issues/586

```
{
  "error": {
    "reason": "Error occurred in Elasticsearch engine: no such index [1]",
    "details": """org.elasticsearch.index.IndexNotFoundException: no such index [1]
For more details, please send request for Json format to see the raw response from elasticsearch engine.""",
    "type": "IndexNotFoundException"
  },
  "status": 404
}
```

**Comment by davidcui-amzn** [comment]
Related: #404 
Need to determine what the best behavior for `Explain` on `SHOW/DESCRIBE` queries are

**Comment by dai-chen** [comment]
I checked other database like MySQL. It doesn't allow EXPLAIN on metadata queries such as SHOW/DESCRIBE. In our case, I think EXPLAIN should only work with queries and disabled for other statements from both frontend and backend.

```
mysql> DESCRIBE accounts;
+-------+------+------+-----+---------+-------+
| Field | Type | Null | Key | Default | Extra |
+-------+------+------+-----+---------+-------+
| age   | int  | YES  |     | NULL    |       |
+-------+------+------+-----+---------+-------+
1 row in set (0.00 sec)

mysql> EXPLAIN DESCRIBE accounts;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DESCRIBE accounts' at line 1
```

MySQL doc: https://dev.mysql.com/doc/refman/8.0/en/explain.html

```
explainable_stmt: {
    SELECT statement
  | TABLE statement
  | DELETE statement
  | INSERT statement
  | REPLACE statement
  | UPDATE statement
}
```

**Comment by davidcui-amzn** [comment]
Looks like Solution Plan will have to come in 2 places:
* `Query Workbench`: Disable `Explain` button on front-end for all non-SELECT queries 
* `SQL/PPL`: Throw Exception in backend with clear error message that `Explain` is not supported for non-SELECT queries

**Comment by davidcui-amzn** [comment]
Fixed in #872
